src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIProxy.cs

```diff
@@ -786,32 +782,44 @@ private bool InferNamedPipesInformation()
 
                 try
                 {
-                    Uri uri = new Uri(_dataSourceAfterTrimmingProtocol);
-                    if (string.IsNullOrEmpty(uri.Host))
+                    string[] tokensByBackSlash = _dataSourceAfterTrimmingProtocol.Split(BackSlashSeparator);
+
+                    // The datasource is of the format \\host\pipe\sql\query [0]\[1]\[2]\[3]\[4]\[5]
+                    // It would at least have 6 parts. 
+                    // Another valid Sql named pipe for an named instance is \\.\pipe\MSSQL$MYINSTANCE\sql\query
+                    if (tokensByBackSlash.Length < 6)
                     {
                         ReportSNIError(SNIProviders.NP_PROV);
                         return false;
                     }
 
-                    string[] absolutePathParts = uri.AbsolutePath.Split(ForwardSlashSeparator);
+                    string host = tokensByBackSlash[2];
```

Comment:
No. Np:\\.\pipe\sql\query would lead to a successful connection but np:\\. \pipe\sql\query doesn't lead to a successful connection in Native SNI. 
We shouldnt trim the pipe parts.